app/jobs/delete_mosaic_job.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+class DeleteMosaicJob < ApplicationJob
+  queue_as :low
+  delegate :query_service, to: :metadata_adapter
+
+  attr_reader :resource_id
+  def perform(resource_id:)
+    @resource_id = resource_id
+    storage_adapter.delete(id: mosaic_id)
+  rescue TileMetadataService::Error
+    false
+  end
+
+  private
+
+    def resource
+      query_service.find_by(id: Valkyrie::ID.new(resource_id))
```

Comment:
Won't this fail if it's backgrounded? The resource will have been deleted.

Reply:
Ah. Right. It will. I've refactored the job to take the storage adapter id as a parameter instead.